UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentAncillaryFiles.cs

```diff
@@ -342,6 +342,19 @@ public async Task<bool> CreateLargeExchangeSetCatalogFile(string batchId, string
                 });
             }
 
+            if (listFulfilmentData != null && listFulfilmentData.Any())
+            {
+                listFulfilmentData = listFulfilmentData.OrderBy(a => a.ProductName).ThenBy(b => b.EditionNumber).ThenBy(c => c.UpdateNumber).ToList();
+
+                List<Tuple<string, string>> orderPreference = new List<Tuple<string, string>> { new Tuple<string, string>("application/s63", "BIN"),
+                    new Tuple<string, string>("text/plain", "ASC"), new Tuple<string, string>("text/plain", "TXT"), new Tuple<string, string>("image/tiff", "TIF") };
+
+                foreach (var listItem in listFulfilmentData)
+                {
+                    CreateLargeExchangeSetCatalogEntry(listItem, orderPreference, catBuilder, salesCatalogueDataResponse, salesCatalogueProductResponse, exchangeSetRootPath, batchId, correlationId);
+                }
+            }
```

Comment:
If listFulfilmentData was null or empty would we still want to Write the Catalogue?

var cat031Bytes = catBuilder.WriteCatalog(path);

Assume this would then be an empty Catalogue? 

Is this a valid outcome?

Reply:
For large exchange sets, our assumption is listFulfilmentData will never be null or empty because we are requesting data via productIdentifiers endpoint. So, as per our testing, this will work fine and the catalog file contains all the product's details. When users request via productVersions endpoint(not for large exchange sets), product identifier with 304 response listFulfilmentData will be empty but catalog and exchange set will get create and catalog file only contains readme file details. If you want, we can add an exception in the case for empty listFulfilmentData.

Reply:
it would be worth defensively coding this.

Reply:
Resolved